travis/composer_check.py

```diff
@@ -0,0 +1,82 @@
+import argparse
+import json
+
+
+def main(orig_file, new_file):
+    """
+    Reads / parses content from the `original_file` and the `new_file`. For each
+    section that we care about ( 'require', 'require-dev', 'extra ), determine whether
+    anything has been added, removed or modified in `new_file`. If there have
+    been changes then we exit w/ a non-zero exit code to indicate that additional
+    validation may be required. If there are no changes detected then we exit w/
+    a zero exit code.
+    """
+    orig_content = read_json(orig_file)
+    new_content = read_json(new_file)
+
+    for key in ['require', 'require-dev', 'extra']:
+        left = value_or_else(orig_content, key)
+        right = value_or_else(new_content, key)
+
+        (added, removed, modified) = dict_compare(left, right)
+        if len(added) > 0 or len(removed) > 0 or len(modified) > 0:
+            exit(1)
+
+
+def read_json(path):
+    """
+    Helper method that opens / parses a json file, returning its contents.
+    :param str path:
+    :return:
+    """
+    with open(path, 'r') as fh:
+        return json.load(fh)
+
+
+def value_or_else(data, key, default=None):
```

Comment:
This capability already exists as a python builtin:

https://docs.python.org/2.7/library/stdtypes.html?highlight=get#dict.get

Reply:
*nod* that's what I get for not googling before writing.